specutils/wcs/specwcs.py

```diff
@@ -169,22 +169,24 @@ def invert(self, dispersion_values):
 
 class Spectrum1DPolynomialWCS(BaseSpectrum1DWCS, polynomial.Polynomial1D):
     """
-    WCS for polynomial dispersion. The only added parameter is a unit,
-    otherwise the same as 'astropy.modeling.polynomial.Polynomial1D'
+    WCS for polynomial dispersion. The only added parameters are a unit,
+    and dc_flag for Log/Linear dispersion.
+    Otherwise the same as 'astropy.modeling.polynomial.Polynomial1D'
     """
-    def __init__(self, degree, unit=None, domain=None, window=[-1, 1],
+    def __init__(self, degree, unit=None, dc_flag=0, domain=None, window=[-1, 1],
```

Comment:
Don't add a logarithmic or linear scaling option to the Polynomial WCS, but if DC flag is set - return a CompositeWCS that has the polynomial WCS and a Logarithmic WCS. For example:

and then you can build your new WCS with:

```python
mywcs = CompositeWCS([my_poly1d_wcs, lambda x: np.pow(10, x)])
````

This way we can repeat exactly the same procedure with the other ways to store data logarithmically.